kratos/utilities/entities_utilities.cpp

```diff
@@ -12,16 +12,199 @@
+        if (rName.find("#") != std::string::npos) { // Templated definition
+            if (rName.find(";") == std::string::npos) { // Must be single definition
+                GenerateTemplatedTypes(rName);
+            } else {
+                KRATOS_ERROR << "Unsupported definition type. Cannot use # and ; together" << std::endl;
+            }
+        } else { // Multiple definition or single definition
+            if (rName.find(";") == std::string::npos) { // Must be single definition
+                GenerateSingleType(rName);
+            } else {
+                GenerateMultipleTypes(rName);
+            }
+        }
+    }
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+template<class TEntity>
+bool EntitityIdentifier<TEntity>::IsInitialized()
+{
+    if (mpPrototypeEntity == nullptr) {
+        return false;
+    } else {
+        // Check if the class is initialized
+        if (mDefinitionType == DefinitionType::Multiple) {
+            // Check size 
+            if (mTypes.size() > 0) {
+                return true;
+            } else {
+                return false;
+            }
+        } else { // Directly check the name
+            return true;
+        }
+    }
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+template<class TEntity>
+const TEntity& EntitityIdentifier<TEntity>::GetPrototypeEntity(typename GeometryType::Pointer pGeometry)
+{
+    switch (mDefinitionType) {
+        case DefinitionType::Single:
+            KRATOS_DEBUG_ERROR_IF_NOT(pGeometry->GetGeometryType() == mpPrototypeEntity->GetGeometry().GetGeometryType()) << "Trying to replace an entity with a different geometry type. Reference entity " << mpPrototypeEntity->GetGeometry().Info() << " vs  " << pGeometry->Info() << "\n Entity info: " << mpPrototypeEntity->Info() << std::endl;
+            return *mpPrototypeEntity;
+        case DefinitionType::Multiple:
+            // We check if the current type is correct or retrieving is required
+            if (pGeometry->GetGeometryType() != mpPrototypeEntity->GetGeometry().GetGeometryType()) {
+                mpPrototypeEntity = (mTypes[pGeometry->GetGeometryType()])->Create(0, pGeometry, nullptr);
+            }
+            return *mpPrototypeEntity;
+        default:
+            KRATOS_ERROR << "Unsupported definition type" << std::endl;
+            break;
+    }
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+template<class TEntity>
+void EntitityIdentifier<TEntity>::PrintData(std::ostream& rOStream) const
+{
+    rOStream << "Name: " << mpPrototypeEntity->Info() << "\n";
```

Comment:
this should be only valid in the single case. In the Multiple case you have a lot of different prototypes

Reply:
True